### PR TITLE
HPCC-15154 Fix inability to write to component.xml files

### PR DESCRIPTION
**Pull request by Michael-Gardner**
Replace reference to $USER variable with invocation of `whoami`.

Signed-off-by: Michael Gardner michael.gardner@lexisnexis.com

### CODE REVIEW

**Comment by Michael-Gardner** [comment]
@xwang2713 This fixes an issue with our calls to configgen.  The calls were being made as root, and subsequently we were having issues with their output files being inaccessible to components running under the hpcc user.

**Comment by xwang2713** [comment]
Looks good

**Comment by Michael-Gardner** [comment]
@richardkchapman Ready for you to take a look

File: initfiles/bash/etc/init.d/hpcc_common.in [hunk 1]
```diff
@@ -302,7 +302,7 @@ configGenCmd() {
     validate_configuration
     configcmd="${configgen_path}/configgen -env ${envfile} -od ${runtime} -id ${componentFile} -c ${compName}"
     log "$configcmd"
-    if [ "${USER}" != "${user}" ]; then
```
Comment (richardkchapman) [inline initfiles/bash/etc/init.d/hpcc_common.in]: There are a lot of references to ${USER} in our scripts. Why is this one different to the others? If it isn't, should we be fixing the others too?

Comment (Michael-Gardner) [inline initfiles/bash/etc/init.d/hpcc_common.in]: I'll dig through our scripts and figure out which one's we need to change

**Comment by Michael-Gardner** [comment]
@richardkchapman I went through and found all the instances of USER that I felt needed to be changed.

**Pull request merged**